src/Nethermind/Nethermind.BlockValidation/Handlers/ValidateBuilderSubmissionHandler.cs

```diff
@@ -0,0 +1,331 @@
+
+        if (lastReceipt.StatusCode != StatusCode.Success)
+        {
+            error = $"Proposer payment failed ";
+            return false;
+        }
+
+        int txIndex = lastReceipt.Index;
+
+        if (txIndex + 1 != block.Transactions.Length)
+        {
+            error = $"Proposer payment index not last transaction in the block({txIndex} of {block.Transactions.Length - 1})";
+            return false;
+        }
+
+        Transaction paymentTx = block.Transactions[txIndex];
+
+        if (paymentTx.To != feeRecipient)
+        {
+            error = $"Proposer payment transaction recipient is not the proposer,received {paymentTx.To} expected {feeRecipient}";
+            return false;
+        }
+
+        if (paymentTx.Value != expectedProfit)
+        {
+            error = $"Proposer payment transaction value is not the expected profit, received {paymentTx.Value} expected {expectedProfit}";
+            return false;
+        }
+
+        if (paymentTx.Data != null && paymentTx.Data.Value.Length != 0)
+        {
+            error = "Proposer payment transaction data is not empty";
+            return false;
+        }
+
+        if (paymentTx.GasPrice != block.BaseFeePerGas)
+        {
+            error = "Malformed proposer payment, gas price not equal to base fee";
+            return false;
+        }
+        error = null;
+        return true;
+    }
+
+    private bool ValidateBlock(Block block, BidTrace message, long registerGasLimit, out string? error)
+    {
+        error = null;
+
+        if (message.ParentHash != block.Header.ParentHash)
+        {
+            error = $"Parent hash mismatch. Expected {message.ParentHash} but got {block.Header.ParentHash}";
+            return false;
+        }
+
+        if (message.BlockHash != block.Header.Hash)
+        {
+            error = $"Block hash mismatch. Expected {message.BlockHash} but got {block.Header.Hash}";
+            return false;
+        }
+
+        if (message.GasLimit != block.GasLimit)
+        {
+            error = $"Gas limit mismatch. Expected {message.GasLimit} but got {block.GasLimit}";
+            return false;
+        }
+
+        if (message.GasUsed != block.GasUsed)
+        {
+            error = $"Gas used mismatch. Expected {message.GasUsed} but got {block.GasUsed}";
+            return false;
+        }
+
+        Address feeRecipient = message.ProposerFeeRecipient;
+        UInt256 expectedProfit = message.Value;
+
+        if (!ValidatePayload(block, feeRecipient, expectedProfit, registerGasLimit, _blockValidationConfig.UseBalanceDiffProfit, _blockValidationConfig.ExcludeWithdrawals, out error))
+        {
+            return false;
+        }
+
+        _logger.Info($"Validated block Hash: {block.Header.Hash} Number: {block.Header.Number} ParentHash: {block.Header.ParentHash}");
+
+        return true;
+    }
+
+    public Task<ResultWrapper<BlockValidationResult>> ValidateSubmission(BuilderBlockValidationRequest request)
```

Comment:
minor: lets move the entry point to the top of the class so it is easy to understand where things start. Other methods should be layout in similar manner (logically close to each other)